presto-raptor/src/main/java/com/facebook/presto/raptor/storage/organization/ShardCompactionManager.java

```diff
@@ -179,28 +169,38 @@ private void discoverShards()
             if (!metadataDao.isCompactionEnabled(tableId)) {
                 continue;
             }
-
-            Table table = metadataDao.getTableInformation(tableId);
-
-            CompactionSetCreator compactionSetCreator = getCompactionSetCreator(table);
-            Set<ShardMetadata> shards = getFilteredShards(entry.getValue(), table);
-
-            addToCompactionQueue(compactionSetCreator, tableId, shards);
+            Collection<OrganizationSet> compactionSets = createCompactionSets(tableId, entry.getValue());
+            compactionSets.stream()
```

Comment:
If you're going to use `.forEach()`, put this one the same line. It looks ugly split on multiple lines like this and doesn't help with readability